### PR TITLE
Core: deprecate jQuery.isFunction

### PR DESCRIPTION
**Pull request by jbedard**
Fixes gh-3609

### Summary ###
Deprecates the public `isFunction`, but it is still used in many places.

### Checklist ###

* [x] All authors have signed the CLA at https://cla.js.foundation/jquery/jquery
* [x] New tests have been added to show the fix or feature works
* [x] Grunt build and unit tests pass locally with these changes
* [x] If needed, a docs issue/PR was created at https://github.com/jquery/api.jquery.com

### CODE REVIEW

**Comment by mgol** [comment]
Could you create Migrate & API tickets referring to this deprecation?

File: src/var/isFunction.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+define( function() {
+	"use strict";
+
+	return function isFunction( obj ) {
+
+        // Support: Chrome <=57, Firefox <=52
```
Comment (dmethvin) [inline src/var/isFunction.js]: Do we even need this function anymore? Currently we're at Chrome 62 and Firefox 57.  **Edit:** Well hecky-darn nevermind, looks like Firefox ESR is [still on 52](https://www.mozilla.org/en-US/firefox/organizations/) until the middle of next year. 😢

**Comment by jbedard** [comment]
Migrate: https://github.com/jquery/jquery-migrate/issues/281
API: https://github.com/jquery/api.jquery.com/issues/1074

**Comment by jbedard** [comment]
Interesting that this one decreases size, unlike #3895. I guess the number of calls (that can now be renamed to a single character) is enough that it can only improve things, gziped or not.
```
   raw     gz Compared to master @ 6c38ebbd47c6b0fa654733819bd5ae36c1ac6c48
  -161     -9 dist/jquery.js
  -430    -19 dist/jquery.min.js
```

**Pull request merged**